dicemix.go

```diff
@@ -523,15 +519,17 @@ func (s *Session) clientRunFail(ctx context.Context, rs *messages.RS) error {
 	}
 	s.freshGen = true
 
+	rsDeadline := time.Now().Add(sendTimeout)
```

Comment:
This looks intentional, but pointing out that it is a change in semantics.  Previously the reveal stage had the entirety of the timeout whereas now it both the failure reporting and the reveal stages have a single deadline.

Reply:
yes, intentional. the runFailed object is very small (only encoded a single struct field that is true), signaling to the server to read the meaty RS message next.  and especially not if the client send deadline is an hour, though i suppose errors would be returned by that time anyways.